clarity_ext/service/clarity_service.py

```diff
@@ -82,3 +86,50 @@ def update_single(self, domain_object, ignore_commit):
                 self.clarity_repository.update(api_resource)
         else:
             raise NotImplementedError("The type '{}' isn't implemented".format(type(domain_object)))
+
+    def get_project_by_name(self, project_name):
+        project_resource = utils.single(self.session.api.get_projects(name=project_name))
+        return ProjectClarityMapper.create_object(project_resource)
+
+    def create_container(self, container, with_samples=False, assign_to=None):
+        """
+        Creates the container and all samples in it. 
+
+        Requires a container and samples that do not have an ID. The samples are interpreted as
+        original samples, not analytes.
+        """
+        if container.id:
+            raise AssertionError("This container already has an ID: {}".format(container))
+        container_type = utils.single(
+                self.session.api.get_containertypes(name=container.container_type))
+        
+        container_res = entities.Container.create(
+                self.session.api, name=container.name, type=container_type)
+        container.id = container_res.id
+
+        if not with_samples:
+            return container
+
+        created_artifacts = list()
+
+        # TODO: Do this in a batch call
+        for well in container.occupied:
+            sample = well.artifact
```

Comment:
A check that sample has not id? (as you did with the container)

Reply:
Right! Forgot that.